### PR TITLE
Add user32 functions and tests

### PR DESCRIPTION
**Pull request by skdltmxn**
I added some functions and related structures of user32.dll as well as tests.

### CODE REVIEW

**Comment by skdltmxn** [comment]
i dont know why, but it looks like AppVeyor doesnt support some functions whose minimum support are Windows 2000...

i commented out them for test.

**Comment by retep998** [comment]
The issue isn't appveyor, it is that for `pc-windows-gnu` Rust depends on import libraries provided by MinGW and unfortunately those import libraries are rather incomplete. For `pc-windows-msvc` it should work fine though because we rely on import libraries provided by Microsoft.

**Comment by skdltmxn** [comment]
OK, I understood thanks. Now, here's another question: how can I test APIs whose minimum supports are newer

**Comment by skdltmxn** [comment]
Oops, wrong button!

**Comment by retep998** [comment]
Could you rebase this PR?

**Comment by skdltmxn** [comment]
I think i rebased it.

**Comment by retep998** [comment]
Nope, that's... an incorrect rebase. You have merge commits and commits that aren't supposed to be part of your PR in your PR.

**Comment by CruzBishop** [comment]
The first time I rebased anything...

Well, let's just say it was ugly. Nice try, hopefully your next one will work okay :+1:

**Comment by skdltmxn** [comment]
OK, i rebased again and it seems right this time...

@CruzBishop Thank you for your kindness :)

File: lib/user32-sys/src/lib.rs [hunk 1]
```diff
@@ -738,8 +847,8 @@ extern "system" {
     // pub fn RegisterSuspendResumeNotification();
     // pub fn RegisterTouchHitTestingWindow();
     // pub fn RegisterTouchWindow();
-    // pub fn RegisterWindowMessageA();
-    // pub fn RegisterWindowMessageW();
+    pub fn RegisterWindowMessageA(lpString: LPCSTR) -> UINT;
+    pub fn RegisterWindowMessageW(lpString: LPCSTR) -> UINT;
```
Comment (retep998) [inline lib/user32-sys/src/lib.rs]: `LPCWSTR`

**Comment by skdltmxn** [comment]
Oops, fixed!
